rest/route/pod.go

```diff
@@ -80,3 +86,59 @@ func (h *podAgentCedarConfig) Run(ctx context.Context) gimlet.Responder {
 	}
 	return gimlet.NewJSONResponse(data)
 }
+
+////////////////////////////////////////////////
+//
+// POST /rest/v2/pods
+
+type podPostHandler struct {
+	body []byte
+	sc   data.Connector
+}
+
+func makePostPod(sc data.Connector) gimlet.RouteHandler {
+	return &podPostHandler{
+		sc: sc,
+	}
+}
+
+func (h *podPostHandler) Factory() gimlet.RouteHandler {
+	return &podPostHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the podID and JSON payload from the HTTP request.
+func (h *podPostHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	h.body = b
+
+	return nil
+}
+
+// Run creates a new resource based on the Request-URI and JSON payload.
+func (h *podPostHandler) Run(ctx context.Context) gimlet.Responder {
+	apiPod := model.APICreatePod{}
+
+	if err := json.Unmarshal(h.body, &apiPod); err != nil {
```

Comment:
I would move the unmarshalling logic into `Parse` since it has to ascertain that the body passes the correct request body format. You can just store the `model.APICreatePod` in the route handler struct.